pkg/test/client_test.go

```diff
@@ -139,6 +155,11 @@ func TestNewClient(t *testing.T) {
 			require.NoError(t, fclient.Status().Patch(context.TODO(), retrieved, depPatch))
 		})
 
+		t.Run("status create fails", func(t *testing.T) {
+			_, retrieved := createAndGetDeployment(t, fclient)
+			require.EqualError(t, fclient.Status().Create(context.TODO(), retrieved, retrieved), "fakeSubResourceWriter does not support create for status")
```

Comment:
Error returned from fakeclient - https://github.com/kubernetes-sigs/controller-runtime/blob/release-0.15/pkg/client/fake/client.go#L1116 
Only substatus resource supported is eviction